pkg/component/autoscaling/clusterautoscaler/cluster_autoscaler.go

```diff
@@ -617,11 +637,45 @@ func (c *clusterAutoscaler) computeShootResourcesData(serviceAccountName string)
 			},
 		}
 	)
+	objects := []client.Object{clusterRole, clusterRoleBinding, role, rolebinding}
 
-	return registry.AddAllAndSerialize(
-		clusterRole,
-		clusterRoleBinding,
-		role,
-		rolebinding,
-	)
+	if workersHavePrioritiesConfigured {
+		configMap, err := c.generatePriorityExpanderConfigMap()
+		if err != nil {
+			return nil, err
+		}
+		objects = append(objects, configMap)
+	}
+	return registry.AddAllAndSerialize(objects...)
+}
+
+func (c *clusterAutoscaler) generatePriorityExpanderConfigMap() (*corev1.ConfigMap, error) {
+	priorities := map[int32][]string{}
+	for _, machineDeployment := range c.machineDeployments {
+		priorities[machineDeployment.Priority] = append(priorities[machineDeployment.Priority], fmt.Sprintf("%s.%s", c.namespace, machineDeployment.Name))
```

Comment:
```suggestion
		priorities[machineDeployment.Priority] = append(priorities[machineDeployment.Priority], fmt.Sprintf("%s\\.%s", c.namespace, machineDeployment.Name))
```

CA expects this to be a regular expression, so shouldn't the `.` be escaped?